doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+            + `application` - Application for which the configuration is meant
+            + `scope` - Application internal scope
+            + `slug` - Entry specific unique slug
+    + filter (string, optional) - Only entries containing the search string in either their path, title, description, author or tags will be added to the output.
+    + filterby (enum[string], optional) - On what fields the filter should be applied to (i.e. what fields to involve in search)
+        + Default: `all`
+        + Members
+            + `all` - All fields: key, title, description, author and tags
+            + `key` - Only the key
+            + `title` - Only the title
+            + `description` - Only the description
+            + `author` - Only the authors name
+            + `tags` - Only tags
+
++ Response 200 (application/json)
+    + Attributes (Entry List)
+
+
+## Get details for entry [GET /database/{organization}/{application}/{scope}/{slug}{?raw}]
+
+Returns detailed information about the entry with the specified key (path). If the optional parameter `raw` is supplied, a raw configuration is returned instead of the normal response.
+
++ Parameters
+    + organization (string, required) - The `organization` who built the `application` for which the configuration of the entry is meant
+    + application (string, required) - The `application` to which the configuration of the entry belongs to
+    + scope (string, required) - The `application` specific `scope` to which the configuration of the entry belongs to
+    + slug (string, required) - The unique identifier of the requested entry
+    + raw (string, optional) - If this parameter is supplied, the API will attempt to return the requested configuration snippet in the given `raw` format
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + data (object) - The actual details about the entry
+            + key (object) - The key and its separate parts
+                + full (string) - The full key (path) of the entry
+                + organization (string) - The organization of the application to which the snippet belongs to
+                + application (string) - The application to which the snippet belongs to
+                + scope (string) - The application internal scope for the configuration snippet
+                + slug (string) - The unique identification slug of the entry
+            + meta (object) - Additional information about the entry
+                + title (string) - The descriptive title of the entry
+                + description (string) - The extended description of the entry
+                + author (string) - The name of the creator of the entry
+                + created_at (number) - A timestamp of the creation date of the entry
+            + value (array[object])
+                + format (string) - A configuration format, e.g. **ini**
```

Comment:
>>>> you get the snippet in all enabled configuration formats

>>>Is this needed? What is the separator between snippets? What happens if some formats fail?

>> Is it needed... definitely not, but still a nice to have in my opinion.

> I do not see a benefit to requesting individual formats

Can't follow you, sorry. From my perspective the API spec is clear in this point and makes sense for both front-ends as well as CLI users. I'll add more explanation to what **raw** means and does, but that's it.

> Maybe an extra tutorial that explains how to use the API with curl?

Something that **could** be done somewhen in future, when the thing is real.

> The snippet service would work much better if there is some kind of auto-detection with score (give the user an option to overwrite if the auto-detection goes awry). So it would be great if you can come up with some heuristics (e.g. count number of =, #, ; and so on) so that we have an auto-detection that works most of the time.

Does sound like the working title of a completely different thesis to me. So long story short: Won't implement.

If there has to be a fallback option to override the format manually, I'd have to implement my currently planned approach additionally eitherway. So I don't see the benefit for me, sorry.